utopia-vscode-extension/src/extension.ts

```diff
@@ -244,13 +218,15 @@ async function doSubscriptionWork(work: SubscriptionWork): Promise<void> {
 
 const SUBSCRIPTION_POLLING_TIMEOUT = 100
 
-async function runPendingSubscriptionChanges(): Promise<void> {
+function runPendingSubscriptionChanges() {
   minimisePendingWork()
   for (const work of pendingWork) {
-    await doSubscriptionWork(work)
+    doSubscriptionWork(work)
   }
 
   pendingWork = []
+
+  // TODO should we still do it like this, or instead follow the pattern used by queueEvents?
```

Comment:
In `utopia-fs.ts` we emit events to inform VS Code of certain file changes, and we batch those together using a 5ms timeout which is reset whenever another event is added to the queue. This just means that when e.g. a ton of new files are added (when initialising the project) we will emit all of those events at the same time. I _think_ we should take the same approach here rather than having a poll that runs every 100ms regardless of the amount of pending work. I'll test that out in a follow up PR.